Add functionalities

vsamy opened the pull request:
Hi,

I needed to add some more functionalities for me. 
It is up to you if you want to merge it.

Here are the modifications:
 - Add primal and dual feasibility tolerance.
 - Add warmstart option.
 - Add inform function that prints solver status.
 - Add possibility to hide/display solver output.
 - Correct some indents.

Vincent S.

Commit:
Add Solver functionalities.
Add primal and dual feasibility tolerance.
Add warmstart option.
Add inform function that prints solver status.
Add possibility to hide/display solver output.

Commit:
Correct indent

haudren commented on src/Gurobi.cpp:
```diff
@@ -59,22 +59,134 @@ const VectorXd& GurobiCommon::result() const
+
+void GurobiCommon::warmStart(int warmStatus)
+{
+	assert(tol >= -1 && tol <= 2);
```
This should be warmStatus. Note that asserts are only triggered in Debug mode.

haudren commented on src/Gurobi.cpp:
```diff
@@ -59,22 +59,134 @@ const VectorXd& GurobiCommon::result() const
 	return X_;
 }
 
+int GurobiCommon::warmStart() const
```
Would it be possible to return a more explicit warmStatus ? Some kind of enum like so maybe:
enum WarmStatus
{
     DEFAULT= -1,
     PRIMAL,
     DUAL,
     NONE
}

haudren reviewed (changes requested):
Thanks for the improvements! If you could just change the warmStatus things, it'll be perfect :+1:

Commit:
Add an enum class for the warmstatus

Commit:
remove useless comment

vsamy commented:
The enum type for warmstatus is a great idea. I have added it and change the test to check the warmstart() functions. I deleted the assert it is no longer needed.

Note that i personnaly like adding assert to test my stuff in debug before going in release mode.

haudren commented:
Nice! Thanks for the changes :)

I was not criticizing the fact that you use asserts, but rather that you should test your asserts in debug mode, so we agree!
